enhancements/security/runtime-class-scc.md

```diff
@@ -0,0 +1,248 @@
+
+```
+type SecurityContextConstraints struct {
+    ...
+	// AllowedRuntimeClasses is a list of names of runtime classes (also called runtime handlers) that can be requested for a pod to run under.
+	// Runtime classes that match this list may be added to the pod request at the pod author's discretion.
+	// To allow all runtime classes, '*' may be specified.
+	// +nullable
+	AllowedRuntimeClasses []string `json:"allowedRuntimeClasses" protobuf:"bytes,26,rep,name=allowedRuntimeClasses"`
+    ...
+}
+```
+Fields in this slice will be logically OR'd. A user must specify an item that only matches one field to be permitted to use that runtime class. This allows an admin to specify a set of runtime classes that a user is allowed to use.
+
+### EnsuredRuntimeClasses
+The API for `EnsuredRuntimeClasses` will be similar in implementation the implementation of [RequiredDropCapabilities](https://github.com/openshift/api/blob/0ba2c3658da6cade27a5066575cd8446eb03914c/security/v1/types.go#L53..L55). `EnsuredRuntimeClasses` will be a slice of strings of runtime classes that must be specified for a specific SCC. Where `AllowRuntimeClasses` is something the user opts into, `EnsuredRuntimeClasses` will gate a request if the runtime class does not match one of the fields.
+
+Similarly to `AllowRuntimeClasses`, `EnsuredRuntimeClasses` will also use a string to specify the set of runtime classes the request must specify:
+```
+type SecurityContextConstraints struct {
+    ...
+	// EnsuredRuntimeClasses is a list of names of runtime classes (also called runtime handlers).     // The runtime class specified in a pod request must match one of the items in this list.
+	// To ensure any runtime class, '*' may be specified.
+	// +nullable
+	EnsuredRuntimeClasses []string `json:"ensuredRuntimeClasses" protobuf:"bytes,26,rep,name=ensuredRuntimeClasses"`
+    ...
+}
+```
+Fields in this slice will also be logically OR'd. A user must specify an item that only matches one field to satisfy the requirement. This allows an admin to specify a set of runtime classes, of which the user must choose at least one.
+
+### Some Examples
+Here are some RuntimeClasses that will be useful for Openshift to support, as well as reasonining behind not allowing everyone to use them:
+- Builder: will need access to /dev/fuse on the host, and a user namespace allocated. While these features are available to all users on the host, an untrusted user could take up all available UIDs/GIDs, preventing builds from happening.
+- High Performance: Allows for pods to request the system is configured for latency sensitive workloads. It allows for CPU load balancing, CPU quota disabling, and IRQ smp load balancing to be disabled. A malicious user could get improper access to the host hardware if allowed to use this runtime class.
```

Comment:
Can we rename this?  I'm hearing it does something with CPU control if CPUs are pinned?

Reply:
how does `openshift-low-latency` work for you?